src/main/java/gregtech/api/capability/impl/CapabilityCompatProvider.java

```diff
@@ -0,0 +1,40 @@
+package gregtech.api.capability.impl;
+
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+
+import javax.annotation.Nullable;
+
+public abstract class CapabilityCompatProvider implements ICapabilityProvider {
+
+    private final ICapabilityProvider upvalue;
+
+    private volatile boolean hasCapabilityLocked;
+    private volatile boolean getCapabilityLocked;
```

Comment:
Not sure how this came about but, there really is no need to lock the provider like its going to be multi-threaded...